frame/buffer.go

```diff
@@ -427,12 +427,11 @@ func (b *Buffer) ReadOpCode() OpCode {
 
 func (b *Buffer) ReadUUID() UUID {
 	if b.err != nil {
-		return UUID{0}
+		return UUID{}
 	}
 
-	s := b.read(16)
-	u := UUID{}
-	copy(u[:], s)
+	var u UUID
+	_, b.err = b.buf.Read(u[:])
```

Comment:
`b.err` shall be set with recordError and wrapped with some context information.

This may change with #98 but for now this is the way to do that.

Reply:
I changed it to error handling from #98